### ISSUE TITLE
docstrings and named parameters for the generated functions

### ISSUE THREAD

**Issue by heyarne** [hellonico/origami#3]
I know the mapping is pretty straight forward and if you have some familiarity with ocv it might not even that much of a benefit, however I tend to explore new clojure APIs in my editor (emacs w/ spacemacs config) by navigating to a symbol and hitting `, h h` which in case of a function displays its signature and the docstring in a minibuffer. Would be super handy for me to have this in origami. Is there any work in this direction? I don't know if named parameters make any more sense then type signatures (they might in liaison w/ spec), but docstrings would help a lot already.

**Comment by hellonico** [comment]
I really like the idea. 
But I can't find anything from the original opencv docs.
https://docs.opencv.org/
The javadoc and other pages are completely empty. 
The one generated when compiling is empty as well.

Any idea of where to find something useful ?

**Comment by heyarne** [comment]
How exactly do you generate the bindings?

**Comment by hellonico** [comment]
I am using some simple java reflection to read the methods and parameter types.
Then output the clojure api methods as text.

**Comment by heyarne** [comment]
No, to be honest I don't have any good idea. In my mind the generated docs of OpenCV were better, and they also seem to have been slightly better at one point. The [docs for 2.4.11](https://docs.opencv.org/java/2.4.11/org/opencv/core/Core.html#absdiff(org.opencv.core.Mat,%20org.opencv.core.Mat,%20org.opencv.core.Mat)) for example have more info. For OpenCV 3 I don't know really, I can't really find anything.

**Comment by geokon-gh** [comment]
Not all the doc strings are empty. I think you just had a unlucky example. For instance:

https://github.com/opencv/opencv/blob/master/modules/imgproc/include/opencv2/imgproc.hpp

Has tons of documentation

But a cursory look tells me it's not going to be fun parsing that into a docstring :)

Sometimes it has something convenient like 
`/** @brief Calculates all of the moments up to the third order of a polygon or rasterized shape.`

While other times it's an 
`/** @overload`
So you have to go find the overload somehow?

And then the enums at the top have a different format with
`//! mode of the contour retrieval algorithm`

I'd ask around and see if anyone has written a converter (I'm seeing javadoc->python-docstring stuff with a quick search) or at least look into some software that reads C++ docstrings and gives you an in-memory mapping

**Comment by Chouffe** [comment]
what about this: https://docs.opencv.org/java/2.4.2/org/opencv/core/Core.html

**Comment by hellonico** [comment]
Some useful text in the latest opencv javadoc
https://docs.opencv.org/4.1.2/javadoc/index.html

https://docs.opencv.org/4.1.2/javadoc/org/opencv/videoio/VideoCapture.html

**Comment by hellonico** [comment]
New Javadoc for 4.3 !
https://docs.opencv.org/master/javadoc/
